install/tools/istio_vm_common.sh

```diff
@@ -0,0 +1,246 @@
+    MIXER_IP=$(kubectl get -n $NS service mixer-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    CA_IP=$(kubectl get -n $NS service istio-ca-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    if [ "${PILOT_IP}" == "" -o  "${ISTIO_DNS}" == "" -o "${MIXER_IP}" == "" ] ; then
+        echo Waiting for ILBs
+        sleep 5
+    else
+        break
+    fi
+  done
+
+  if [ "${PILOT_IP}" == "" -o  "${ISTIO_DNS}" == "" -o "${MIXER_IP}" == "" ] ; then
+    echo "Failed to create ILBs"
+    exit 1
+  fi
+
+  #/etc/dnsmasq.d/kubedns
+  echo "server=/svc.cluster.local/$ISTIO_DNS" > kubedns
+  echo "address=/istio-mixer/$MIXER_IP" >> kubedns
+  echo "address=/mixer-server/$MIXER_IP" >> kubedns
+  echo "address=/istio-pilot/$PILOT_IP" >> kubedns
+  echo "address=/istio-ca/$CA_IP" >> kubedns
+
+  CIDR=$(gcloud container clusters describe ${K8SCLUSTER} ${GCP_OPTS:-} --format "value(servicesIpv4Cidr)")
+  echo "ISTIO_SERVICE_CIDR=$CIDR" > cluster.env
+}
+
+# Get an istio service account secret, extract it to files to be provisioned on a raw VM
+# Params:
+# - service account -  defaults to istio.default or SERVICE_ACCOUNT env
+# - service namespace - defaults to current namespace.
+function istio_provision_certs() {
+  local SA=${1:-${SERVICE_ACCOUNT:-istio.default}}
+  local NS=${2:-${SERVICE_NAMESPACE:-}}
+
+  if [[ -n "$NS" ]] ; then
+    NS="-n $NS"
+  fi
+
+  kubectl get $NS secret $SA -o jsonpath='{.data.cert-chain\.pem}' |base64 -d  > cert-chain.pem
+  kubectl get $NS secret $SA -o jsonpath='{.data.root-cert\.pem}' |base64 -d  > root-cert.pem
+  kubectl get $NS secret $SA -o jsonpath='{.data.key\.pem}' |base64 -d  > key.pem
+}
+
+
+# Install required files on a VM and run the setup script.
+#
+# Must be run for each VM added to the cluster
+# Params:
+# - name of the VM - used to copy files over.
+# - optional service account to be provisioned (defaults to istio.default)
+# - optional namespace of the service account and VM services, defaults to SERVICE_NAMESPACE env
+# or kube config.
+function istioBootstrapVM() {
+ local NAME=${1}
+
+ local SA=${2:-${SERVICE_ACCOUNT:-istio.default}}
+ local NS=${3:-${SERVICE_NAMESPACE:-}}
+
+  istio_provision_certs $SA
+
+  local ISTIO_FILES=${ISTIO_FILES:-.}
+
+ # Copy deb, helper and config files
+ # Reviews not copied - VMs don't support labels yet.
+ istioCopy $NAME \
+   kubedns \
+   *.pem \
+   cluster.env \
+   $ISTIO_FILES/istio_vm_setup.sh \
+   $ISTIO_FILES/istio-proxy-envoy.deb \
+   $ISTIO_FILES/istio-agent.deb \
+   $ISTIO_FILES/istio-auth-node-agent.deb \
+
+ # Run the setup script.
+ istioRun $NAME "sudo bash -c -x ./istio_vm_setup.sh"
+}
+
+
+# Helper functions for the main script
+
+# If Istio was built from source, copy the artifcats to the current directory, for use
+# by istioProvisionVM
+function istioCopyBuildFiles() {
+  local ISTIO_IO=${ISTIO_BASE:-${GOPATH:-$HOME/go}}/src/istio.io
+  local ISTIO_FILES=${ISTIO_FILES:-.}
```

Comment:
it is not super clear to me what that variable is/does exactly in this function, is it a destination directory, a source dir, a set of files...

Reply:
Added more docs.